docs-core/src/main/java/com/sismics/docs/core/dao/ScoreDao.java

```diff
@@ -32,4 +37,35 @@ public String create(Score score, String userId) {
         
         return score.getId();
     }
+
+    /**
+     * Get all scores on a document.
+     * 
+     * @param documentId Document ID
+     * @return List of scores
+     */
+    public List<ScoreDto> getByDocumentId(String documentId) {
+        EntityManager em = ThreadLocalContext.get().getEntityManager();
+        StringBuilder sb = new StringBuilder("select s.SCORE_ID_C, s.SCORE_SKILLS_C, s.SCORE_EXPERIENCE_C, s.SCORE_TRANSCRIPTGPA_C, SCORE_MATCH_C, SCORE_CREATEDATE_D from T_SCORE c");
+        sb.append(" where s.SCORE_IDDOC_C = :documentId and s.SCORE_DELETEDATE_D is null ");
+        sb.append(" order by s.SCORE_CREATEDATE_D asc ");
+        Query q = em.createNativeQuery(sb.toString());
+        q.setParameter("documentId", documentId);
+        @SuppressWarnings("unchecked")
+        List<Object[]> l = q.getResultList();
+        
+        List<ScoreDto> scoreDtoList = new ArrayList<>();
+        for (Object[] o : l) {
+            int i = 0;
+            ScoreDto scoreDto = new ScoreDto();
+            scoreDto.setId((String) o[i++]);
+            scoreDto.setSkills((String) o[i++]);
+            scoreDto.setExperience((String) o[i++]);
+            scoreDto.setTranscriptGpa((String) o[i++]);
+            scoreDto.setMatch((String) o[i++]);
```

Comment:
Could you confirm if the scores (skills, experience, transcriptGpa, match) should be Integer or String type? The set functions return Integer types but I'm not sure if the `(String)` is casting the values to String.

Reply:
The set functions take in String parameters here and cast to integers.